### DISCUSSION TITLE
Does not work HwGM_PTZ

### DISCUSSION THREAD

**Discussion by riogrande75** [Arkady23/V380-Web#11] [General]
I tried _httpd_GM35SCOMMON_V2_v20210530.zip_ with my V380 camera with following specs:
**SW: AppGM35S_CARD_V3.0.6.9_20151202
FW: HwGM_PTZ_V4.0_20151018**

Unfortunately script from sd-card did not start automatically, so I had to run "startup.sh" manually (via UART console):
```
/mnt/sdcard/ark-add-on # ./startup.sh
/mnt/sdcard/ark-add-on/update.sh: line 12: tar: not found
/mnt/sdcard/ark-add-on/update.sh: line 13: tar: not found
/mnt/sdcard/ark-add-on/update.sh: line 14: tar: not found
/mnt/sdcard/ark-add-on # ./startup.sh: line 5: /mnt/sdcard/ark-add-on/bin/httpd: not found
```
[V380bootup.txt](https://github.com/Arkady23/V380-Web/files/6808081/V380bootup.txt)
Is there any chance that I get this device supported? (Maybe simply tar is not available...).

**Comment by Arkady23** [comment]
Yes, it is possible. Because you have access to the console. I suspect that you have a different hwname. Through the console, you can see the hwname. Indeed, tar was not found. Maybe it is necessary to write the path to tar? /sbin/tar or /bin/tar? If you can't find the tar module, I can give it from my camera.

**Comment by riogrande75** [comment]
How can I find out the hwname?

No - I cannot find tar it in the system (searched with _find / -name tar_).
Can you pls. send me your tar binary?

**Reply by Arkady23** [reply]
[ark-add-on.zip](https://github.com/Arkady23/V380-Web/files/6808380/ark-add-on.zip)
This is an unpacked version, it does not need tar. Just put everything inside the archive in the root of the sd card. Wait a minute.... I'm looking for a tar module. I'll post it later.

**Reply by Arkady23** [reply]
[tar.zip](https://github.com/Arkady23/V380-Web/files/6808400/tar.zip)
If it doesn't work, then I can try to change it to local libraries.

**Reply by Arkady23** [reply]
> How can I find out the hwname?

I know that the hwname was found through the uart. Perhaps in order for it to be displayed on the console, it is necessary to place the initial patch on the sd card.

**Reply by riogrande75** [reply]
No - unfortunately tar does not work...
```
mnt/sdcard/ark-add-on # ./tar

./tar: symbol '__stdin': can't resolve symbol
```

**Reply by riogrande75** [reply]
I can use the "unpacked" version to get it working.
But how can I make the "patch" work to start it automatically every bootup?

**Reply by Arkady23** [reply]
In the last message, I prepared for you patch.sh. Did you launch it? Has it worked? If not, is there a protocol, what is the error when executing it? You can make everything run by itself every time. But only I need to have feedback from you. With the tar module, this is a big problem, since tar is often used. I have one idea so far - to put tar on sd in a special patch for your camera.

**Reply by riogrande75** [reply]
There is no output on the console when I start patch.sh and webserver does not start.
Only when I manually start /mnt/sdcard/ark-add-on/startup.sh webservice comes up fine. After reboot everything gone until I start it again.
See full bootlog, patch.sh, startup.sh, reboot, startup.sh.
[V380_Patch_full.log](https://github.com/Arkady23/V380-Web/files/6813678/V380_Patch_full.log)
Any improvement very welcome.

**Reply by Arkady23** [reply]
Give these 3 commands, please else:
```
cat /mnt/mtd/mvconf/factory_const.ini
cat /mnt/mtd/vg_boot.sh
find / -name *.sh
```

**Reply by riogrande75** [reply]
Here you go.
[v380abfr.txt](https://github.com/Arkady23/V380-Web/files/6821639/v380abfr.txt)

**Reply by Arkady23** [reply]
You have everything installed as it should be. But there is a note on the composition of the sd card. You need to delete everything. Because you have both a patch and an add-on there. Patch, all its files must be deleted. Leave only the folder **RecFiles**. Delete everything else and write down the contents of the unpacked add-on again from https://github.com/Arkady23/V380-Web/files/6808380/ark-add-on.zip
The httpd server must start. If it does not run, then we will put an intermediate output in the startup file to understand the situation.

**Reply by riogrande75** [reply]
Thx a lot for your effort.
Unfortunately even after wiping sdcard and placing only files from the last add-on the http-deamon does not start automatically.
If I start ip with `/mnt/sdcard/ark-add-on/startup.sh` it works.
What diagnosis data will help you to find the cause?

**Reply by Arkady23** [reply]
Replace the original startup.sh with this. I wonder what the terminal will give out.
[startup.zip](https://github.com/Arkady23/V380-Web/files/6829420/startup.zip)
Just don't run it yourself. It should start on its own.

**Reply by riogrande75** [reply]
Attached logs (log.txt and terminal output).
[log.txt](https://github.com/Arkady23/V380-Web/files/6829703/log.txt)
[v380_startup2.txt](https://github.com/Arkady23/V380-Web/files/6829706/v380_startup2.txt)

**Reply by Arkady23** [reply]
Did you launch this yourself startup.sh? And you need a download protocol when you turn on the camera. In any case, you do not need to run it yourself, otherwise it will be unclear.

**Reply by riogrande75** [reply]
See, this seems to be the problem with my cam: `startup.sh` does not get executed at start.
So off course there is no log.txt when i start up without any manual action.

**Reply by Arkady23** [reply]
No. I checked your log. /mnt/mtd/vg_boot.sh is being launched, and there is a launch in it /mnt/sdcard/ark-add-on/startup.sh.

**Reply by Arkady23** [reply]
I can assume that the sd is not mounted yet at the time of launch. But we need to check.

**Reply by Arkady23** [reply]
By the way, does telnet start? Have you checked it?

**Reply by riogrande75** [reply]
Yes, telnet is started.
Can you pls. create a patch that  /mnt/mtd/vg_boot.sh is not checking and simply starting startup.sh?
Something like
```/mnt/mtd/nvipcstart.sh &
sd=/mnt/sdcard/ark-add-on
$sd/startup.sh &
/ #
```

**Reply by Arkady23** [reply]
So you already have this. You have shown me vg_boot.sh. But so far, the protocol when the camera is turned on has not been sent with a test one startup.sh. Therefore, I do not know what you have and what to do.

**Comment by riogrande75** [comment]
Thx - webserver working! :-)
But I have to start it manually every time. Any Idea on how I could get this fixed too?
Here the ouput - I guess hwname is not included here:
![grafik](https://user-images.githubusercontent.com/28924474/125444685-690a0f03-beb6-486e-ba7f-29cf08f84d2b.png)
What I should look for in the boot messages?

**Reply by Arkady23** [reply]
Try to run this command file through the console patch.sh. And then it will reboot.
[patch.zip](https://github.com/Arkady23/V380-Web/files/6808552/patch.zip)
Put patch.sh on the sd card.

**Reply by Arkady23** [reply]
Did the tar program work for you?

**Reply by Arkady23** [reply]
> What I should look for in the boot messages?

No need to search. Everything is already visible. Hwname GM35SCOMMON_V2 version 3000609

**Reply by riogrande75** [reply]
No, does not work either.
If I do not start the _httpd_ manually, it does'nt run after startup.
Sorry for the late answer.

**Reply by Arkady23** [reply]
I think that in addition to tar, you will still have a lot of things missing in the system. Try updating the web server with the latest update.

**Reply by Arkady23** [reply]
[bin.zip](https://github.com/Arkady23/V380-Web/files/7896240/bin.zip)
There is a tar here

**Reply by Arkady23** [reply]
It's bad that there is no feedback. I think it's probably necessary for you to check the sd, and the delay is 20 seconds.

**Reply by riogrande75** [reply]
I'll give it a shot today.

**Reply by riogrande75** [reply]
Today I created a new sd-card with _httpd_GM35SCOMMON_V2_v20210530.zip_ and _patch.sh_ from above.
Started up my cam, after a while I ran manually _patch.sh_ -> no output.
Then I shut down camera and extracted files from your last post (bin.zip) to sdcard/bin and sdcard/ark-add-on/bin dir. 
Started camera again, but nothing get's started.
Attached log from procedure.
[V380_210122.txt](https://github.com/Arkady23/V380-Web/files/7912524/V380_210122.txt)

Can you pls. create a full archive with all required files for debugging my ZCAM C380? I really want to have it running on my device.
